### PR TITLE
 Feature wifi module into release

### PR DESCRIPTION
**Pull request by PatrickDekker98**
We an interface for esp32 was made together with an test interface for external connection

### CODE REVIEW

File: code/main.cpp [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
-        hwlib::wait_ms(1000);
+
+    auto sclk = hwlib::target::pin_out(hwlib::target::pins::d9);
+    auto mosi = hwlib::target::pin_out(hwlib::target::pins::mosi);
+    auto miso = hwlib::target::pin_in(hwlib::target::pins::miso);
+    auto ss = hwlib::target::pin_out(hwlib::target::pins::d10);
+
+    //configure spi bus with cs 0 = pin 10
+    //configure spi bus with spi mode 1
+    //configure with difider 42 ( 84 / 42 = 2Mhz )
```
Comment (TonyvanderKrogt) [inline code/main.cpp]: It is divider

File: code/esp_code/tcp_client.c [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+static void tcp_client_task(void *pvParameters) {
+    //char rx_buffer[128];
+    char addr_str[128];
+    int addr_family;
+    int ip_protocol;
+    
+    while (1) {
+
+#ifdef CONFIG_EXAMPLE_IPV4
+        struct sockaddr_in dest_addr;
```
Comment (TonyvanderKrogt) [inline code/esp_code/tcp_client.c:88]: According the code style structs end with _s

File: code/esp_code/tcp_client.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        dest_addr.sin_addr.s_addr = inet_addr(HOST_IP_ADDR);
+        dest_addr.sin_family = AF_INET;
+        dest_addr.sin_port = htons(PORT);
+        addr_family = AF_INET;
+        ip_protocol = IPPROTO_IP;
+        inet_ntoa_r(dest_addr.sin_addr, addr_str, sizeof(addr_str) - 1);
+#else // IPV6
+        struct sockaddr_in6 dest_addr;
```
Comment (TonyvanderKrogt) [inline code/esp_code/tcp_client.c:96]: According the code style structs end with _s

File: code/headers/mock_external.hpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+         * @brief set a vector of frames to be received
+         *
+         * @param frames , vector of frames to be set
+         */
+        void set_receive_frames(const std::vector<r2d2::frame_external_s> frames);
+
+        /**
+         * @brief set a reaction lambda
+         *
+         * @param reaction , a reaction lamda wich can react
```
Comment (TonyvanderKrogt) [inline code/headers/mock_external.hpp]: Wich is which

**Comment by TonyvanderKrogt** [comment]
For the test you can maybe explain what it is testing so people know what they are testing
In the same file you use abbreviations like retvec. According the code style it is not allowed for abbreviations to prevent misunderstandings

The comments are great with a few spelling errors. Change a few things and its ok

File: code/main.cpp (hunk 1 above)
Comment (PatrickDekker98) [reply code/main.cpp]: fixed in new commit

File: code/headers/mock_external.hpp (hunk 4 above)
Comment (PatrickDekker98) [reply code/headers/mock_external.hpp]: What do you mean?

File: code/esp_code/tcp_client.c (hunk 3 above)
Comment (PatrickDekker98) [reply code/esp_code/tcp_client.c:96]: These are not declared within or divined within my code, they are part of the esp library

File: code/esp_code/tcp_client.c (hunk 2 above)
Comment (PatrickDekker98) [reply code/esp_code/tcp_client.c:88]: See comment below

File: code/headers/mock_external.hpp (hunk 4 above)
Comment (TonyvanderKrogt) [reply code/headers/mock_external.hpp]: The correct spelling is which instead of wich.

**Review by TonyvanderKrogt — APPROVED** [review]
All changes are checked and changed if necessary

**Pull request merged**